File: Java/Linked List/Singly Circular Linked List/SinglyCL01.java
/*
=====================================================================

Name            : Singly Linked List - Node Creation and Linking
Program Number  : 001
Description     : This program demonstrates the creation of two
                  nodes in a Singly Linked List and links them
                  together using the next reference.

                  The program:
                  - Creates two nodes
                  - Stores data in each node
                  - Connects first node to second node
                  - Displays node data

Concepts Used   : Singly Linked List, Node Creation,
                  Object Instantiation, Reference Variables,
                  Data Members, Link Establishment

Input           : None

Output          : Data stored in linked list nodes

Date            : July 2026
Language        : Java
Author          : Atharv Tushar Bhosale

=====================================================================
*/

class node
{
    public int data;
    public node next;
}

class SinglyCL01
{
    public static void main(String A[])
    {
        node first = null;
        node second = null;

        first = new node();
        second = new node();

        first.data = 11;
        first.next = second;

        second.data = 21;
        second.next = null;

        System.out.println(first.data);
        System.out.println(second.data);
    }
}

/*
=====================================================================

Sample Output :

11
21

=====================================================================

Classes Used :

1. node
2. SinglyCL01

=====================================================================

Observation :

- Two nodes are created dynamically.
- First node is linked to second node.
- Data can be accessed using object references.
- This is the basic foundation of a Singly Linked List.

=====================================================================
*/

File: C++/Linked List/Singly Circular Linked List/SinglyCL01.cpp
/*
=====================================================================

Name            : Singly Circular Linked List - Node Structure Size
Program Number  : 001

Description     : This program demonstrates the basic node structure
                  used in a Singly Circular Linked List.

                  Each node contains:
                  - Data field
                  - Pointer to next node

                  The program displays the size occupied by the node
                  structure in memory.

Concepts Used   : Structure, Self Referential Structure,
                  Memory Layout, sizeof Operator

Input           : None

Output          : Size of node structure

Date            : July 2026
Language        : C++
Author          : Atharv Tushar Bhosale

=====================================================================
*/

#include<iostream>
using namespace std;

#pragma pack(1)
struct node
{
    int data;
    struct node* next;
};

typedef struct node NODE;
typedef struct node* PNODE;

int main()
{
    cout<<"Size of node is : "<<sizeof(NODE)<<endl;

    return 0;
}

/*
=====================================================================

Output :

Size of node is : 12

=====================================================================

*/
